src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1553,10 +1716,63 @@ private void CancelPendingOutboundRequest(object state)
             }).Forget();
         }
 
+        private void TraceLocalMethodAdded(string rpcMethodName, MethodSignatureAndTarget targetMethod)
+        {
+            Requires.NotNullOrEmpty(rpcMethodName, nameof(rpcMethodName));
+
+            if (this.TraceSource.Switch.ShouldTrace(TraceEventType.Information))
```

Comment:
Are these `ShouldTrace()` checks necessary? Or an optimization? I thought `TraceEvent()` would already check the switch condition.

Reply:
So did I. But when I added tracing to the `MultiplexingStream` inside Nerdbank.Streams, I found that tracing is _very_ slow even with it turned off unless I added explicit checks like this.

Reply:
slow and/or GC pressure heavy.